packages/image_picker/image_picker/android/src/main/AndroidManifest.xml

```diff
@@ -1,7 +1,10 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-  package="io.flutter.plugins.imagepicker">
-   <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
-   <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
+    package="io.flutter.plugins.imagepicker">
+
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
```

Comment:
Can someone point me towards documentation that talks about why we have either of these permissions declared? As mentioned [here](https://developer.android.com/training/data-storage) (see the first row in the table), no permissions are needed for accessing the cache directory. I think these permissions are over-reaching and it'd be nice to drop them.

Reply:
Right, the same goes for any API level of course, not just 29+. `WRITE_EXTERNAL_STORAGE` should not be needed at all anymore. 

I cannot actually find any evidence that the `READ_EXTERNAL_STORAGE` permission is required for `Intent.ACTION_GET_CONTENT`, but it seems like a logical permission to request as we are reading images written by other apps, from external storage.
I have tried removing the permission, but it ends up throwing `PlatformException(photo_access_denied, The user did not allow photo access., null, null)`, without prompting the user.

Therefore I would propose just removing the `WRITE_EXTERNAL_STORAGE` permission entirely and leaving `READ_EXTERNAL_STORAGE` as is.

Reply:
Can you tell me more about your issue with that platform exception? I put together [this sample](https://github.com/btrautmann/image_picker_sample) to prove you don't need that permission (in the sample I use the get content action successfully without it). If I can reproduce the issue it may help me understand why it's needed.

Reply:
Sorry @BeMacized I just had another look at the code. You're running into that Exception because you likely removed the permission declaration from the manifest but didn't remove the request for it from the code, which isn't allowed. If you remove it in both places you should be able to fetch the image without issue.

Reply:
Ah I see! that makes things a lot more clear, thanks for the hint. 
I've made the changes so that the read permission is no longer required either.

Reply:
I know I'm probably being a pest @BeMacized but it's worth noting the plugin also doesn't need to request the camera permission because it's using the `ACTION_VIDEO_CAPTURE` and `ACTION_IMAGE_CAPTURE` intent actions. You can see that explained [here](https://developer.android.com/training/permissions/evaluating#take-photo). I figure since this is now a major release and the PR addresses permissions, it may be nice to do a sweep and remove them all?

FWIW, the reason I'm calling these all out is because it's nice for client apps to not request these permissions needlessly. It helps build trust with our users, so I'd love to see these go away if possible :)

Reply:
If we could drop that one too that would indeed be excellent! I'll definitely take a look at it, although I can't guarantee it will also land together with this PR.